core/engine.go

```diff
@@ -131,12 +131,14 @@ func (e *Engine) Run(ctx context.Context) error {
 
 	collectorwg := sync.WaitGroup{}
 	collectorctx, collectorcancel := context.WithCancel(context.Background())
-	if e.Collector != nil {
-		collectorwg.Add(1)
-		go func() {
-			e.Collector.Run(collectorctx)
-			collectorwg.Done()
-		}()
+	if len(e.Collectors) > 0 {
+		for _, collector := range e.Collectors {
+			collectorwg.Add(1)
+			go func(collector lib.Collector) {
```

Comment:
I'm not sure we need goroutines here, see [my comment](https://github.com/loadimpact/k6/pull/617/files#r188240264) about `Collect()` in the kafka PR.

Reply:
So I didn't add it, I'm just using what was already there ... did you want me to work on removing it?  Should that be in scope for this PR?

But I can play with it ... maybe it would help address the issue of why I switched from `Run()` to `Collect()` for sending the samples to Kafka that I was talking about in the Kafka PR.

Reply:
Argh, my bad, I misread the diff and the collector method called, ignore my comment entirely. It's exactly how it's supposed to be and your changes are perfectly fine.